extensions/amp-social-share/0.1/validator-amp-social-share.protoascii

```diff
@@ -72,11 +54,31 @@ tags: {  # <amp-social-share>
   attrs: {
     name: "type"
     mandatory: true
-    value_regex: "(twitter|facebook|pinterest|linkedin|gplus|email)"
+  }
+  attrs: {
+    name: "data-share-endpoint"
+    value_url: {
+      allowed_protocol: "ftp"
+      allowed_protocol: "http"
+      allowed_protocol: "https"
+      allowed_protocol: "mailto"
+      # Whitelisting additional commonly observed third party
+      # protocols which should be safe
+      allowed_protocol: "sms"
+      allowed_protocol: "viber"
+      allowed_protocol: "whatsapp"
```

Comment:
1. Yes.
2. Added some example with 'j a v a s c r i p t : '. Since this exercises the same code as the anchor checks, the validation is already quite covered with the tests in here:
https://github.com/ampproject/amphtml/blob/master/validator/testdata/feature_tests/urls.html

The gist is that the validator will detect the protocol and compare it to a whitelist.